docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Understanding tokens"
+description: "Understand how large language models (LLMs) use tokens to analyze semantic relationships and generate natural language outputs"
+author: haywoodsloan
+ms.topic: concept-article
+ms.date: 04/05/2024
+
+#customer intent: As a .NET developer, I want understand how large language models (LLMs) use tokens to analyze semantic relationships and generate natural language outputs, so I may better use LLMs in my .NET projects.
+
+---
+
+# Understanding tokens
+
+Tokens are the words or character sets that large language models (LLMs) decompose text into. During training, an LLM analyzes the semantic relationships between these tokens, such as how commonly they're used together or if they're used in similar contexts. Once trained, an LLM uses those patterns and relationships to generate a sequence of output tokens based on the input sequence.
+
+## What are tokens?
+
+The first step when training an LLM or inputting a prompt is tokenization. The text is broken down into a set of strings, known as _tokens_. These tokens can be distinct words, character sets, or combinations of words and punctuation. The set of unique tokens a model has been trained on is referred to as its _vocabulary_.
+
+For example, consider the following sentence:
+
+> I heard a dog bark loudly at a cat
+
+This text could be tokenized as:
+
+- I
+- heard
+- a
+- dog
+- bark
+- loudly
+- at
+- a
+- cat
+
+Only the first occurrence of the "a" token will be added to the model's vocabulary. All other occurrences can be represented by that first token.
+
+With a sufficiently large set of training text, a vocabulary of many thousands of tokens can be compiled.
+
+## How are tokens determined?
+
+The specific tokenization method varies by LLM. Some common tokenization methods include:
+
+- Word tokenization (text is split into individual words based on a delimiter)
+- Character tokenization (text is split into individual characters)
+- Subword tokenization (text is split into partial words or character sets)
+
+For example, the GPT models, developed by OpenAI, use _Byte-Pair Encoding_ (BPE) which is a type of subword tokenization. OpenAI provides [a tool to visualize how text will be tokenized](https://platform.openai.com/tokenizer).
+
+There are benefits and disadvantages to each tokenization method:
+
+**Smaller tokens (character or subword tokenization)**
```

Comment:
Consider redoing as a table or columns.

Reply:
reformatted into a table, had to use some inline HTML to put a list inside the table

Reply:
For bullets in a table, I just use - and `<br/>`. Anything else might get flagged at build time.

Reply:
Thanks for the tip. I revised the table to just have `<br>` tags